File: order-service/src/main/java/com/ecommerce/order/entity/OrderItem.java
package com.ecommerce.order.entity;

import com.ecommerce.shared.enums.ProductColor;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "order_items")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Order order;

    @ManyToOne
    private ShoppingCart shoppingCart;

    private Long productId;

    private String productName;

    private String productImage;

    private String size;

    @Enumerated(EnumType.STRING)
    private ProductColor color;

    private int quantity;

    private double price;

}

File: mobile_app/lib/models/order_item.dart
class OrderItem {
  final int id;
  final int productId;
  final String productName;
  final String productImage;
  final String size;
  final String color;
  final double price;
  final int quantity;
  final bool selected;

  OrderItem({
    required this.id,
    required this.productId,
    required this.productName,
    required this.productImage,
    required this.size,
    required this.color,
    required this.price,
    required this.quantity,
    required this.selected,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: json['id'],
      productId: json['productId'],
      productName: json['productName'],
      productImage: json['productImage'],
      size: json['size'],
      color: json['color'],
      price: json['price'],
      quantity: json['quantity'],
      selected: json['selected'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'productId': productId,
      'productName': productName,
      'productImage': productImage,
      'size': size,
      'color': color,
      'price': price,
      'quantity': quantity,
      'selected': selected,
    };
  }
}
